source/blog/2018-01-18-ovirt-admin-bookmarks.md

```diff
@@ -0,0 +1,33 @@
+---
+title: oVirt 4.2.2 web admin UI browser bookmarks
+author: awels,
+tags: oVirt
+date: 2018-01-19 12:00:00 CET
+---
+
+oVirt web admin UI now allows the user to bookmark all entities and searches using their browser.
+
+READMORE
+
+## Synchronizing URL with application state
+
+Whenever you select a detail view in the application, the browser URL is now updated to match the selected entity. For instance if you have a VM named MyVM and you click on the name to see the details, the URL of the browser will go to #vms-general;name=MyVM. If you switch to lets say the network interfaces tab the URL in your browser will switch to #vms-network_interfaces;name=MyVM. Changing entity or changing location will keep the browser URL synchronized. This allows you to use your browsers bookmark functionality to store a link to that VM.
+
+## Direct linking to entities
+
+As a complementary functionality you can pass arguments to places that will execute some functionality based on the type of argument you have passed in. The following types are available:
+
+* SEARCH, is for main views only, this allows you to pre populate the search string used in the search bar.
+* NAME, most entities are uniquely named and you can use their name in a detail view to go directly to that named entity.
+* DATACENTER, quota and networks are not uniquely named, but are unique combined with their associated data center, to link directly to either you need to specify NAME and DATACENTER.
+* NETWORK, VNIC profiles are not uniquely named, but need both DATACENTER and NETWORK to be specified to directly link to it. 
+
+This functionality allows the bookmarks to work and take you directly to the entity/search in your bookmark. This also allows external applications to direct link to entities in the webadmin.
```

Comment:
Please mention that this also works even if the user isn't already logged in, I'd replace the last sentence with following:

"
If the user isn't already logged in, they will be redirected to oVirt SSO login page and then back to the desired place in the application. This allows external applications to directly link to entities in web admin UI.
"